modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/SnapshotManagerComponent.java

```diff
@@ -353,13 +356,26 @@ protected void beforeCreateDocument(DocumentModel doc) {
         snapDoc = session.saveDocument(snapDoc);
         snapDoc = session.move(snapDoc.getRef(), session.getParentDocumentRef(tmp.getRef()), null);
 
+        commitOrRollbackAndRestartTransaction();
```

Comment:
Maybe add a comment just above to explain why we are committing and waiting: we need to wait for the move that is done asynchronously at the MongoDB level.